### PR TITLE
awsebscsiprovisioner chart version 0.3.5 -> aws-ebs-csi-driver 0.5.0

### PR DESCRIPTION
**Pull request by sebbrandt87**
**What type of PR is this?** 
Feature

**What this PR does/ why we need it**:
- adds the new awsebscsiprovisioner chart 0.3.5
- aws-ebs-csi-driver 0.5.0
- updated compatible CSI container images
- includes more values to be setable

**Which issue(s) this PR fixes**:
[D2IQ-64990]
[D2IQ-64992]

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

[D2IQ-64990]: https://jira.d2iq.com/browse/D2IQ-64990
[D2IQ-64992]: https://jira.d2iq.com/browse/D2IQ-64992

### CODE REVIEW

**Comment by sebbrandt87** [comment]
tested on new konvoy clusters and it is provisioning as it should.
also encryption is working and snapshotting also, if you will create the snapshotclass to be created as well. Ticket for that here: [D2IQ-66015]

[D2IQ-66015]: https://jira.d2iq.com/browse/D2IQ-66015

File: addons/awsebscsiprovisioner/0.5.x/awsebscsiprovisioner-1.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  requires:
+    - matchLabels:
+        kubeaddons.mesosphere.io/name: defaultstorageclass-protection
+  kubernetes:
+    minSupportedVersion: v1.15.6
+  cloudProvider:
+    - name: aws
+      enabled: true
+  chartReference:
+    version: 0.3.5
```
Comment (shaneutt) [inline addons/awsebscsiprovisioner/0.5.x/awsebscsiprovisioner-1.yaml]: Did we need to bump this?

Comment (joejulian) [reply addons/awsebscsiprovisioner/0.5.x/awsebscsiprovisioner-1.yaml]: Ah, :hankey:. I didn't see this until after I merged.

**Comment by joejulian** [comment]
@sebbrandt87 Since your comments suggest that there are user-facing changes (encryption and snapshotting, at least), please update the release-notes. These are used for creating the actual release.

**Comment by samba** [comment]
@joejulian @sebbrandt87 did the release notes get updated, per Joe's request?

**Comment by sebbrandt87** [comment]
sure @samba see "Does this PR introduce a user-facing change?:" <-- this is the content of the release notes on the docs sites, it gets generated from this

**Pull request merged**